harness/tests/cli/test_experiment.py

```diff
@@ -44,20 +45,48 @@ def mock_det_auth(user: str = "test", master_url: str = "http://localhost:8888")
 
 
 @unittest.mock.patch("determined.common.api.authentication.Authentication")
-def test_wait_unstable_network(
+def test_wait_doesnt_throw_exception_on_master_504(
     auth_mock: unittest.mock.MagicMock,
 ) -> None:
     auth_mock.return_value = mock_det_auth()
     user = "user1"
+    experiment_id_flaky = 1
     with api_server.run_api_server(
         credentials=(user, "password1", "token1"), ssl_keys=None
     ) as master_url:
-        args = CliArgs(master=master_url, experiment_id=1)
+        args = CliArgs(master=master_url, experiment_id=experiment_id_flaky)
         determined.cli.experiment.wait(args)
 
 
 @unittest.mock.patch("determined.common.api.authentication.Authentication")
-def test_wait_stable_network(
+def test_wait_waits_until_longrunning_experiment_is_complete(
+    auth_mock: unittest.mock.MagicMock,
+) -> None:
+    auth_mock.return_value = mock_det_auth()
+    user, password, token = "user", "password1", "token1"
+    api_server_session = _session.Session(
+        master=f"{api_server.DEFAULT_HOST}:{api_server.DEFAULT_PORT}",
+        user=user,
+        auth=None,
+        cert=None,
+    )
+    experiment_id_longrunning = 2
+
+    with api_server.run_api_server(
+        credentials=(user, password, token), ssl_keys=None
+    ) as master_url:
+        args = CliArgs(master=master_url, experiment_id=experiment_id_longrunning)
+        determined.cli.experiment.wait(args)
+
+        fetched_experiment = experiment.ExperimentReference(
+            experiment_id_longrunning, api_server_session
+        )._get()
+
+    assert fetched_experiment.state == bindings.experimentv1State.STATE_COMPLETED
```

Comment:
question: does this call and assert fail if they're made before the wait?

Reply:
The call succeeds and the assertion fails. Would you like to see a comment about that or the whole thing differently expressed?

Reply:
oh, that checks out I missed this change https://github.com/hamidzr/determined/blob/9aee839b045c296d97177f11fb832b7bc2af3ad8/harness/tests/common/api_server.py#L132-L133

no need for more comments, thanks.

Reply:
I'm right there with you. This interaction is not at all obvious. I think it's worth taking another look at the design of `api_server` so that the fixtures can be injected from the file with the tests.

... But it sounds like you're good with not doing that right now, and that sounds good to me, too.